R/addins.R

```diff
@@ -97,77 +97,43 @@ get_rstudio_context <- function() {
   rstudioapi::getActiveDocumentContext()
 }
 
-
-# Dedicated binding for package styling addin. Simple wrapper calling style_pkg
-# with the selected addins style.
-style_package <- function() {
-  communicate_addins_style()
-  style_pkg(style = get_addins_style_fun())
-}
-
-
-# `match.fun`-like utility covering "ns::name".
-exported_value_rx <- "^([^:]+)::([^:]+)$"
-is_exported_value <- function(x) {
-  rlang::is_scalar_character(x) && grepl(exported_value_rx, x)
-}
-extract_exported_ns <- function(x) {
-  sub(exported_value_rx, "\\1", x)
-}
-extract_exported_name <- function(x) {
-  sub(exported_value_rx, "\\2", x)
-}
-match_fun <- function(x) {
-  if (is_exported_value(x)) {
-    x <-
-      getExportedValue(
-        extract_exported_ns(x),
-        extract_exported_name(x)
-      )
-  }
-  match.fun(x)
-}
-
-# Binding for style-setting addin.
+#' Ask the user to supply a style
+#'
+#' @keywords internal
 prompt_style <- function() {
-  current_style <- get_addins_style()
+  current_style <- get_addins_style_name()
   new_style <-
     rstudioapi::showPrompt(
       "Select a style",
       "Enter the name of a style function, e.g. `styler::tidyverse_style`",
       current_style
     )
-  if (!is.null(new_style)) {
-    set_addins_style(new_style)
+  parsed_new_style <- tryCatch(
+    eval(parse(text = new_style)),
+    error = function(e) e
```

Comment:
We can actually stop if there is an error in `eval(parse(...))` with a custom and more informative message, instead of relying on the non-function message:
```suggestion
    error = function(e) {
      stop("The selected style \"", new_style, "\" is not valid: ", e$message)
    }
```
This way for e.g. `"styler::untidyverse_style"` you get something like 
```
The selected style "styler::untidyverse_style" is not valid: 'untidyverse_style' is not an exported object from 'namespace:styler' 
```

Reply:
Yes we can be verbose about that, true.